com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/topology.py

```diff
@@ -49,9 +49,14 @@ def sink(self, func):
 
     def filter(self, func):
         """
-        Filters tuples from a stream using the supplied function.
-        For each tuple on the stream the function is called passing
-        the tuple, if the function return evalulates to true the
+        Filters tuples from a stream using the supplied callable 'func'.
+        The callable input must be either 
+        1) the name of a top-level function defined using 'def' that takes a single parameter for the tuple, or
+        2) an instance of a top-level callable class that implements the method '__call__(self, tuple)'.  Using a
```

Comment:
Changed the comment to read "functions defined at the top level of a module".  This is the same terminology used in Python doc here:
https://docs.python.org/3/library/pickle.html#what-can-be-pickled-and-unpickled

All I want to say here is that we don't support nested functions, since an inner function is local the outer function and thus not visible by name to PythonC.